internal/controllers/gateway/gateway_utils.go

```diff
@@ -670,3 +666,71 @@ func isTLSSecretValid(secret *corev1.Secret) bool {
 	}
 	return true
 }
+
+func routeAcceptedByGateways(routeNamespace string, parentStatuses []RouteParentStatus) []k8stypes.NamespacedName {
+	gateways := []k8stypes.NamespacedName{}
+	for _, routeParentStatus := range parentStatuses {
+		gatewayNamespace := routeNamespace
+		parentRef := routeParentStatus.ParentRef
+		if (parentRef.Group != nil && *parentRef.Group != gatewayV1beta1Group) ||
+			(parentRef.Kind != nil && *parentRef.Kind != "Gateway") {
+			continue
+		}
+		if parentRef.Namespace != nil {
+			gatewayNamespace = string(*parentRef.Namespace)
+		}
+		if lo.ContainsBy(routeParentStatus.Conditions, func(condition metav1.Condition) bool {
+			return condition.Type == string(gatewayv1beta1.RouteConditionAccepted) &&
+				condition.Status == metav1.ConditionTrue
+		}) {
+			gateways = append(gateways,
+				k8stypes.NamespacedName{
+					Namespace: gatewayNamespace,
+					Name:      string(parentRef.Name),
+				})
+		}
+	}
+	return gateways
+}
+
+// Get all the HTTPRoutes and update the listeners' attachedRoutes field.
+// At this point we take into account HTTPRoutes only, as they are the
+// only routes in beta.
+func getAttachedRoutesForListener(ctx context.Context, client client.Client, listener gatewayv1beta1.Listener, gateway gatewayv1beta1.Gateway) (int32, error) {
+	httpRouteList := gatewayv1beta1.HTTPRouteList{}
+	if err := client.List(ctx, &httpRouteList); err != nil {
+		return 0, err
+	}
+	var attachedRoutes int32
+	for _, route := range httpRouteList.Items {
+		route := route
+		if acceptedByGateway := func() bool {
+			for _, g := range routeAcceptedByGateways(route.Namespace, route.Status.Parents) {
+				if gateway.Namespace == g.Namespace && gateway.Name == g.Name {
+					return true
+				}
+			}
+			return false
+		}(); !acceptedByGateway {
+			continue
+		}
```

Comment:
To consider

```suggestion
		_, acceptedByGateway := lo.Find(
			routeAcceptedByGateways(route.Namespace, route.Status.Parents),
			func(g k8stypes.NamespacedName) bool {
				return gateway.Namespace == g.Namespace && gateway.Name == g.Name
			},
		)
		if !acceptedByGateway {
			continue
		}
```

Reply:
I think that `lo` is useful in many cases, but in contexts like this one decreases code readability instead of improving it (very personal opinion).

Reply:
My sole puropose for this comment to try to steer you in the way of restructuring the `if` block here a bit. It might be a matter of preference though. For me, the way it's done now, one might get lost trying to find where the condition ends and where is does the inner block of `if` starts.

Reply:
Ok, I get your point. I changed it 👍